### PR TITLE
integrates checkstyle plugin to check against Google Java Codestyle

### PR DESCRIPTION
**Pull request by timoe**
## Description
Maven build now checks code using Googles Java code style. Refactored codebase for matching that codestyle.

## Related Issue
Fixes https://github.com/adobe/S3Mock/issues/70

## Tasks

- [x] I have signed the [CLA](http://adobe.github.io/cla.html).
- [x] I have written tests and verified that they fail without my change.

---

@agudian: Please review, merge + rebase.

### CODE REVIEW

File: server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java [hunk 1]
```diff
@@ -673,12 +690,13 @@ public InitiateMultipartUploadResult initiateMultipartUpload(
   /**
    * Lists all in-progress multipart uploads.
    *
-   * http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadListMPUpload.html
+   * <p>http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadListMPUpload.html</p>.
```
Comment (agudian) [inline server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java]: Dangling `.` at the end

File: server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java [hunk 2]
```diff
@@ -714,7 +732,7 @@ public ListMultipartUploadsResult listMultipartUploads(@PathVariable final Strin
   /**
    * Aborts a multipart upload for a given uploadId.
    *
-   * http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadAbort.html
+   * <p>http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadAbort.html</p>.
```
Comment (agudian) [inline server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java]: Dangling `.`

File: testsupport/testng/src/test/java/com/adobe/testing/s3mock/testng/S3MockListenerXmlConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    final InputStream uploadFileIs = new FileInputStream(uploadFile);
+    final String uploadHash = HashUtil.getDigest(uploadFileIs);
+    final String downloadedHash = HashUtil.getDigest(s3Object.getObjectContent());
+    uploadFileIs.close();
+    s3Object.close();
+
+    Assert.assertEquals(uploadHash, downloadedHash,
+        "Up- and downloaded Files should have equal Hashes");
+  }
+}
```
Comment (agudian) [inline testsupport/testng/src/test/java/com/adobe/testing/s3mock/testng/S3MockListenerXmlConfigurationTest.java]: Newline at eof?

File: server/src/main/java/com/adobe/testing/s3mock/S3MockApplication.java [hunk 4]
```diff
@@ -207,8 +201,8 @@ public int getHttpPort() {
    *
    * @param keyRef A KMS Key Reference
    */
-  public void registerKMSKeyRef(final String keyRef) {
-    kmsKeyStore.registerKMSKeyRef(keyRef);
+  public void registerKmsKeyRef(final String keyRef) {
```
Comment (agudian) [inline server/src/main/java/com/adobe/testing/s3mock/S3MockApplication.java]: This is part of the public API that people are using. Let’s not change this. Maybe we should ignore that rule completely, as some of the other changes to comply with it seem a bit forced (e.g. for SSLContext, or the renaming of the *IT classes to *It.

Comment (timoe) [reply server/src/main/java/com/adobe/testing/s3mock/S3MockApplication.java]: Right breaking the interface is of course a bad idea. I personally like the style not having to or more upper case letter, also for IntegrationTests. However, I revert the interface changes and the *IT stuff. Unfortunately we will end up not using Google Standard Checks but create and maintain our own

**Comment by timoe** [comment]
udpated (+squashed). Please verify additional broken interfaces.

**Comment by agudian** [comment]
I've just played around with it a bit:
* I've set consoleOutput to `false` to not repeat the output that would be reported properly (colored and with shortened paths) anyway.
* Use the built-in google_checks.xml and deactivated the rule `AbbreviationAsWordInName` (doesn't require us to maintain a customized google_checks.xml file, but would allow names like `getMMORPGScore()` -  we might want to just keep an eye on that in PRs)
* `build-config` module can be removed again
* configured the `execution` part in the parent-pom, so the other poms only need to add the following:
```xml
  <build>
    <plugins>
      <plugin>
        <artifactId>maven-checkstyle-plugin</artifactId>
      </plugin>
    </plugins>
  </build>``` 

Here's what I had in the parent-pom:
```xml
        <plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-checkstyle-plugin</artifactId>
          <version>3.0.0</version>
          <configuration>
            <consoleOutput>false</consoleOutput>
            <failOnViolation>true</failOnViolation>
            <violationSeverity>info</violationSeverity>
            <includeTestSourceDirectory>true</includeTestSourceDirectory>
            <violationIgnore>google_checks.xml</configLocation>
            <violationIgnore>AbbreviationAsWordInName</violationIgnore>
          </configuration>
          <dependencies>
            <dependency>
              <groupId>com.puppycrawl.tools</groupId>
              <artifactId>checkstyle</artifactId>
              <version>${checkstyle.version}</version>
            </dependency>
          </dependencies>
          <executions>
            <execution>
              <id>check-style</id>
              <phase>validate</phase>
              <goals>
                <goal>check</goal>
              </goals>
            </execution>
          </executions>
```

WDYT?

**Comment by agudian** [comment]
> Please verify additional broken interfaces.

Looks all good 👍

**Comment by timoe** [comment]
> WDYT?

I found that option for configuring checkstyle-pljugin but: What about IDE integration? IDEA (here Checkstyle-Plugin) does not pick this configuration as a rule so you'll always have that mismatch here :/. You can't even configure custom suppressions for build-in configurations. So, that's the whole thing here.

We can simply drop that idea of having checks IDE-integrated with the benefit of not having a custom config. Thoughts?

**Pull request merged**